feat(skore): Add data_source="both" to ComparisonReport metrics frames

Adrien-1997 opened the pull request:
Adds support for `data_source="both"` in ComparisonReport metric displays.

Train and test data are now stored in a single long DataFrame with an explicit
`data_source` column and plotted on separate subplots when requested, avoiding
ambiguous comparisons.

Applies to:
- Precision-Recall (Closes #2154)
- ROC (Closes #2155)
- Prediction Error (Closes #2156)

<img width="1280" height="480" alt="pe_train_test_plot" src="https://github.com/user-attachments/assets/a525763d-bab4-47c7-914b-32665cbc9958" />
<img width="1280" height="480" alt="pr_train_test_plot" src="https://github.com/user-attachments/assets/42db6f6b-ba06-44e6-b168-a22b62bbe200" />
<img width="1280" height="480" alt="roc_train_test_plot" src="https://github.com/user-attachments/assets/b3a2cec1-2f9b-44f6-a5ae-ff009daac0a1" />

Closes #1874

GaetandeCast commented on skore/tests/unit/displays/prediction_error/test_comparison_estimator.py:
```diff
@@ -322,3 +322,47 @@ def test_title(pyplot, linear_regression_with_train_test):
+@pytest.fixture
+def linear_regression_comparison_report(linear_regression_with_train_test):
+    """Fixture providing a ComparisonReport with two linear regression estimators."""
+    estimator, X_train, X_test, y_train, y_test = linear_regression_with_train_test
```
Let's move this to `conftest.py` where the other fixtures are defined.

GaetandeCast reviewed (changes requested):
Last change then it's looking good for me!

Adrien-1997 commented:
I’ve moved the comparison regression fixture to `conftest.py` with the other shared fixtures and updated the tests accordingly.

Also wanted to say I really enjoyed this review process, thanks a lot @GaetandeCast !

GaetandeCast reviewed (approved):
You're welcome! Let's see if @glemaitre wants to add something here before we merge.
